R/brood_check.R

```diff
@@ -757,10 +757,10 @@ check_values_brood <- function(Brood_data, var) {
   } else if(var %in% c("LayDate", "LayDate_observed")) {
 
     ref <- Brood_data %>%
-      dplyr::filter(!is.na(!!rlang::sym(var))) %>%
+      dplyr::filter(!is.na(!!rlang::sym(var)) & !is.na(Species)) %>%
       dplyr::group_by(BreedingSeason) %>%
-      # Transform dates to julian days to calculate quantiles
-      dplyr::mutate(!!paste0(var, "_julian") := lubridate::yday(!!rlang::sym(var))) %>%
+      # Transform dates to Julian days (while accounting for year) to calculate quantiles
+      dplyr::mutate(!!paste0(var, "_julian") := lubridate::yday(!!rlang::sym(var)) + (lubridate::year(!!rlang::sym(var)) - BreedingSeason) * 365) %>%
```

Comment:
Perhaps an easier approach would be to just find the difference between dates. That way you don't have to multiply by 365 (which could cause some weirdness with leapyears). Suggestion below, not much prettier but I think a bit more robust.

```suggestion
      dplyr::mutate(!!paste0(var, "_julian") := as.numeric(!!rlang::sym(var) - lubridate::ymd(paste(BreedingSeason, "1", "1", sep = "-")) + 1)) %>%
```